contracts/Pool.sol

```diff
@@ -102,18 +110,20 @@ contract Pool {
     /// @notice Returns amount of received funds available for collection
     /// by the sender of the message
     /// @return collected The available amount
-    function collectable() public view returns (uint256) {
+    function collectable() public view returns (uint128) {
         Receiver storage receiver = receivers[msg.sender];
         uint64 collectedCycle = receiver.nextCollectedCycle;
         if (collectedCycle == 0) return 0;
+        uint256 currFinishedCycle = block.number / cycleBlocks;
+        if (collectedCycle > currFinishedCycle) return 0;
         uint256 collected = 0;
```

Comment:
We don’t we declare this as uint128? We coerce it when we return anyways.

Reply:
This is something I'm hesitant about in general, we can either:

- A. Use the smallest possible possible integers fitting necessary data. This makes reasoning about code a bit easier, but requires Solidity to artificially maintain the lowered precision. All storage and operations on the EVM are performed on 256-bit values.
- B. Use 256-bit values everywhere possible. This obscures invariants about the possible values of variables, but aligns with the EVM architecture.

I honestly have no idea, which one is better. Now I'm thinking that maybe you're right and we should focus on core readability and let the optimizer take care of uglification and necessary casts?

Reply:
This answers my question and I’d be ok with keeping it as it is but I’d still prefer to avoid unecessarry casts and use the type that we intend to use.

Reply:
Fixed, reduced number of casts and slimmed down integer types to better match what they can actually contain